### PR TITLE
Example: add example running sim and reco for CLIC

### PR DESCRIPTION
**Pull request by andresailer**
This adds text on how to run simulation and reconstruction for CLIC, at the moment the k4run instructions still require the manual modification of the steering file. What would be the best way to add command line arguments for this?

### CODE REVIEW

File: examples/clic.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+git clone https://github.com/iLCSoft/CLICPerformance
+cd CLICPerformance/clicConfig
+```
+
+## Simulation
+
+Now we can already simulate a few events
+```bash
+ddsim --compactFile $LCGEO/CLIC/compact/CLIC_o3_v14/CLIC_o3_v14.xml \
```
Comment (vvolkl) [inline examples/clic.md:21]: This requires `$DD4HEP` to be set to the examples directory of the dd4hep installation. In spack this is not (yet) the case. There are really too many of these variables!

Comment (vvolkl) [reply examples/clic.md:21]: Also, in the key4hep stack `dd4hep+lcio` is still missing, so this fails. I'll add that.

Comment (andresailer) [reply examples/clic.md:21]: Thanks for checking.
I have this diff in my local clicPerformance that fixes the problem you observe
```diff
-SIM.physics.pdgfile = os.path.join( os.environ.get("DD4HEP"), "DDG4/examples/particle.tbl" )
+SIM.physics.pdgfile = os.path.join( os.environ.get("DD4hepINSTALL"), "examples/DDG4/examples/particle.tbl" )
```

**Review by fdplacido — APPROVED** [review]

File: examples/clic.md (hunk 1 above)
Comment (vvolkl) [reply examples/clic.md:21]: I just tried with `source /cvmfs/clicdp.cern.ch/iLCSoft/lcg/97/nightly/x86_64-centos7-gcc8-opt/init_ilcsoft.sh` but then `$LCGEO` is not set. Would be great if it works with both stacks.

File: examples/clic.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+## Reconstruction
+
+### Reconstruction with Marlin
+
+To run the reconstruction with ``Marlin``
+```bash
+Marlin clicReconstruction.xml \
+       --InitDD4hep.DD4hepXMLFile=$LCGEO/CLIC/compact/CLIC_o3_v14/CLIC_o3_v14 \
```
Comment (vvolkl) [inline examples/clic.md:35]:
```suggestion
       --InitDD4hep.DD4hepXMLFile=$LCGEO/CLIC/compact/CLIC_o3_v14/CLIC_o3_v14.xml \
```

File: examples/clic.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+       --global.LCIOInputFiles=ttbar.slcio \
+       --global.MaxRecordNumber=3
+```
+
+### Reconstruction with with Gaudi
+
+We can convert the ``xml`` steering file to a Gaudi steering file
+```bash
+convertMarlinSteeringToGaudi.py clicReconstruction.xml  > clicReconstruction.py
```
Comment (vvolkl) [inline examples/clic.md:44]: I still need to install the latest version of gmp for this.

File: examples/clic.md (hunk 1 above)
Comment (andresailer) [reply examples/clic.md:21]: set `lcgeo_DIR` from the spack recipe?
Doing something like
`${LCGEO-${lcgeo_DIR}}` probably just works in bash

File: examples/clic.md (hunk 3 above)
Comment (andresailer) [reply examples/clic.md:44]: https://github.com/key4hep/k4-spack/pull/62

File: examples/clic.md (hunk 1 above)
Comment (andresailer) [reply examples/clic.md:21]: https://github.com/iLCSoft/CLICPerformance/pull/132 for the particle.tbl issue

**Comment by vvolkl** [comment]
I checked again with a fresh install, and to run `clicReconstruction.py`, the command line options are indeed needed. I'll have to adapt the k4run script to allow this - since all the algorithms are of type MarlinProcessorWrapper with only a Parameters property, it currently exposes it only once (for the last algorithm defined I think). I remember that I did this on purpose, since in the fcc config files this situation is somewhat unusual, and not having to deal with the algorithmname in the parameters reduced the complexity quite a bit. Opened https://github.com/key4hep/K4FWCore/issues/17 to keep track of it.

**Comment by vvolkl** [comment]
running `gaudirun.py clicReconstruction.py` (after changing lines 
``: `read.Files = ["ttbar.slcio"]`
18: `            "DD4hepXMLFile", "/cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/lcgeo-0.16.6-n2cn5dmwe4zttcgltxamnbfzq26kofie/share/lcgeo/compact/CLIC/compact/CLIC_o3_v14/CLIC_o3_v14.xml", END_TAG,`
1385: `EvtMax   = 3,`
 )

I get an error:

````
rimaryVertices_res PrimaryVertices_RP PrimaryVertices_res_RP RefinedVertices RefinedVertices_RP 
[ MESSAGE "Output_DST"] 	LCIOOutputFile:  Output_DST.slcio
[ MESSAGE "Output_DST"] 	LCIOWriteMode:  WRITE_NEW
[ MESSAGE "Output_DST"] -------------------------------------------------
EventLoopMgr      WARNING Unable to locate service "EventSelector" 
EventLoopMgr      WARNING No events will be processed from external input.
[ MESSAGE "EventNumber"]  ===== Run  :       0  Event:       0
[ WARNING "MyDDMarlinPandora"] Failed to extract track collection: SiTracks_Refitted, lcio::DataNotAvailableException: LCEventImpl::getCollection: collection not in event:SiTracks_Refitted
=================== WARNING ===================
static method CellIDDecoder::setDefaultEncoding
does nothing anymore. Use the constructor to 
specify your encoding string !!
===============================================
[ WARNING "MergeRP"]  MergeCollections::processEvent : incompatible parameter vector sizes : InputCollections: 3 <->  InputCollectionIDs 0
[ WARNING "MergeRP"]  MergeCollections::processEvent : standard numbering (0,1,2,...) used.
[ WARNING "MergeClusters"]  MergeCollections::processEvent : incompatible parameter vector sizes : InputCollections: 3 <->  InputCollectionIDs 0
[ WARNING "MergeClusters"]  MergeCollections::processEvent : standard numbering (0,1,2,...) used.
[ MESSAGE0 "MyRecoMCTruthLinker"]  processEvent 0  - 0
MyRecoMCTruthLi...  FATAL  Standard std::exception is caught 
MyRecoMCTruthLi...  ERROR lcio::DataNotAvailableException: LCEventImpl::getCollection: collection not in event:MCPhysicsParticles
EventLoopMgr        FATAL .executeEvent(): Standard std::exception thrown by MyRecoMCTruthLinker
EventLoopMgr        ERROR lcio::DataNotAvailableException: LCEventImpl::getCollection: collection not in event:MCPhysicsParticles
EventLoopMgr      WARNING Execution of algorithm MyRecoMCTruthLinker failed
[ ERROR "MyTrackChecker"] Collection SiTracks_Refitted is unavailable

```

@fdplacido  have you seen this before?

**Comment by vvolkl** [comment]
I forgot about 478:                         `  "MaxRecordNumber", "3", END_TAG,`
but the error remains the same

**Comment by andresailer** [comment]
@vvolkl you need to enable the two processors with `# Config.OverlayFalse`  in the algList

**Comment by fdplacido** [comment]
Check if you uncommented the necessary parts at the bottom of the script, for the algorithm list.

**Comment by andresailer** [comment]
Also those for `Config.ConformalTracking`

**Comment by vvolkl** [comment]
Thanks, but still no:

```
[ WARNING "MergeClusters"]  MergeCollections::processEvent : standard numbering (0,1,2,...) used.
[ MESSAGE0 "MyRecoMCTruthLinker"]  processEvent 0  - 0
MyRecoMCTruthLi...  FATAL  Standard std::exception is caught 
MyRecoMCTruthLi...  ERROR lcio::DataNotAvailableException: LCEventImpl::getCollection: collection not in event:MCPhysicsParticles
EventLoopMgr        FATAL .executeEvent(): Standard std::exception thrown by MyRecoMCTruthLinker
EventLoopMgr        ERROR lcio::DataNotAvailableException: LCEventImpl::getCollection: collection not in event:MCPhysicsParticles
EventLoopMgr      WARNING Execution of algorithm MyRecoMCTruthLinker failed
[ ERROR "MyTrackChecker"] Collection SiTracksMCTruthLink is unavailable
```
Where is `SiTracksMCTruthLink` produced?

**Comment by andresailer** [comment]
Yes, there is still a bug in the last tag of the overlay processor.
```
OverlayFalse.Parameters = [
                           "Verbosity", "MESSAGE", END_TAG,
                           "BackgroundFileNames", "", END_TAG,
...
```
add the empty string to the BackgroundFileNames for the moment.
Need to tag the Overlay package

**Comment by vvolkl** [comment]
Thanks! Now it works. Do we want to add the necessary edits on the page here (I can write a few `sed` commands that do the substitions ), or include a fixed `clicReconstruction.py` with GMPWrapper?

**Comment by andresailer** [comment]
The overlay thing should be fixed with https://github.com/key4hep/k4-spack/pull/65

I would just explicitly mention the other changes for the commented algList lines. 
But in addition the sed commands would be nice for copy and paste solution.

Latter we could try to improve the converted to add boilerplate command line flags for the if constructs?

**Comment by vvolkl** [comment]
All the updated versions should be in cvmfs now. Will give this another try.

**Comment by vvolkl** [comment]
Works now with https://github.com/andresailer/key4hep-doc/pull/1 !

**Comment by vvolkl** [comment]
Just investigating the build failure.

**Pull request merged**